### PR TITLE
MT-4085 bulk order

### PR DESCRIPTION
**Pull request by yulia-dnistrian**
Description:

Linting:

- [x] No linting errors

Tests:

- [ ] E2E tests
- [x] Manual tests
- [x] Accessibility tests (no new react-axe errors in console)

Documentation:

- [ ] Requires documentation updates
- [ ] Requires Storybook component updates

### CODE REVIEW

File: src/AppHeader.tsx [hunk 1]
```diff
@@ -55,7 +57,18 @@ export const AppHeader: React.FC = () => {
...
             <CartIcon className="appheader__carticon" />
           </button>
+          {showCartPopup && (
+            <div className="appheader__cartpopup">
+              <p>{t('cart-popup-info', {quantity: quantity.toString()})}</p>
+              <button className="epbtn" onClick={handleCartModal}>{t('view-cart')}</button>
+            </div>
+          )}
         </div>
+        {isLoggedIn && (
```
Comment (shaunmaharaj) [inline src/AppHeader.tsx:67]: Confirmed with @BonnieEP, we should be displaying this for guest users too. We can remove this check.

File: src/BulkOrder.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        .catch(error => {
+          setBulkError(error.errors[0].detail);
+          setShowLoader(false);
+          console.error(error);
+        });
+    }
+  });
+
+  useEffect(() => {
+    const bulkOrderItems:any = values.productSKU
```
Comment (shaunmaharaj) [inline src/BulkOrder.tsx:47]: Thank you for implementing this

File: src/app-state.ts [hunk 3]
```diff
@@ -378,12 +380,20 @@ function useCartItemsState() {
...
-      setTotalPrice(res.meta.display_price.without_tax.formatted)
+      const itemQuantity = res.data.reduce((sum, { quantity }) => sum + quantity, 0);
+      setQuantity(itemQuantity - count);
+      setCount(itemQuantity);
+      setTotalPrice(res.meta.display_price.without_tax.formatted);
+      if (!showCartPopup && itemQuantity - count !== 0) {
+        setShowCartPopup(true);
+        setTimeout(() => {
+          setShowCartPopup(false);
+        }, 3200);
```
Comment (shaunmaharaj) [inline src/app-state.ts:391]: Did we get the value for this delay from UX? Let's just verify with Bonnie if this is fast/slow enough

File: src/QuickOrder.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    itemsArr[index] = {...itemsArr[index]};
+    arg.map(el => {
+      return itemsArr[index][Object.keys(el)[0]] = Object.values(el)[0];
+    });
+    setItems(itemsArr);
+  };
+
+  const handleBlur = (index:number, code:string) => {
+    if (code === '') return;
+    getProductById(code)
```
Comment (shaunmaharaj) [inline src/QuickOrder.tsx:35]: We should be able to pass product ID's directly to the form without the need for this check. Is this check purely done to validate if the product exists? If so, let's discuss with Yury and Bonnie on a better experience.

We'll have to update this anyway to accept skus vs product IDs.

**Comment by shaunmaharaj** [comment]
@yulia-dnistrian Can you please make the following updates from Bonnie's comments in this PR:

- we should we show SKUs of products on PDP and Product List pages, similar as we did for EPC Ref Store
- we can show bulk/quick order for non-logged in users as well
- The form with no products added for quick order should show zero beside each input. When the user adds a valid SKU, the SKU added with a default quantity of 1

**Comment by shaunmaharaj** [comment]
@yulia-dnistrian can you please make some final updates based on Bonnie's UX audit?

- Update the “View cart” link hover state to be an underline (currently this is a pale green background)
- If 1 item has been added to the cart please update the message to say “item” not “items”  (use “items” if 2 or more items are added at a time) and also use “has” for all added messages
- In cart the “Remove” button needs to be updated. The default colour should be Teal (#167A92) with a underline hover state

**Comment by BonnieEP** [comment]
The feature is working as expected except for a few caveats. Thank you :thank-you: @yulia-dnistrian 
Some updates needed:

1. Update the “View cart” link hover state to be an underline (currently this is a pale green background)
2. If 1 item has been added to the cart please update the message to say “item” not “items”  (use “items” if 2 or more items  are added at a time) and also use “has” for all added messages
3. In cart the “Remove” button needs to be updated. The default colour should be Teal (#167A92) with a underline hover state

**Comment by BonnieEP** [comment]
> @yulia-dnistrian can you please make some final updates based on Bonnie's UX audit?
> 
> * Update the “View cart” link hover state to be an underline (currently this is a pale green background)
> * If 1 item has been added to the cart please update the message to say “item” not “items”  (use “items” if 2 or more items are added at a time) and also use “has” for all added messages
> * In cart the “Remove” button needs to be updated. The default colour should be Teal (#167A92) with a underline hover state

@yulia-dnistrian if you have any questions about this please let me know :)

**Review by BonnieEP — APPROVED** [review]

**Comment by BonnieEP** [comment]
Screen Shot 2020-10-26 at 11 58 55 AM
@yulia-dnistrian the error after I've cleared the input seems to be sticking in the UI. Can you update this some that the error once corrected disappears?

**Review by shaunmaharaj — APPROVED** [review]

**Pull request merged**